### PR TITLE
MM-14673 Search "in:[username]" returns results in GMs

### PR DESCRIPTION
**Pull request by a-arias**
#### Summary
Write an automated test using `cypress` for "Search "in:[username]" returns results in GMs" 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10482

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

**Comment by a-arias** [comment]
@saturninoabril  This is ready to review

**Comment by thekiiingbob** [comment]
@a-arias Looks like some of the checks are failing. can you resolve?

File: e2e/cypress/integration/messaging/search_group_message_spec.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            cy.get('.more-modal__name').contains(text).click();
+
+            //# save values
+            cy.get('#saveItems').click();
+
+            //# searching 'lucho' text and hitting enter
+            cy.get('#post_textbox').clear().type('lucho').type('{enter}');
+
+            //# writing "in:" text in search input
+            cy.get('#searchBox').type('in:');
```
Comment (thekiiingbob) [inline e2e/cypress/integration/messaging/search_group_message_spec.js]: Everything up to this point can most likely be API calls to setup the data, and then we are only concerned with typing in the search box and asserting results. You'll have to most likely start by making a group channel (https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1group%2Fpost) and then having one of the users post to that channel.

**Comment by jasonblais** [comment]
Hey @a-arias, thanks for your contributions! Let us know if we can help clarify the feedback above

File: e2e/cypress/integration/messaging/search_group_message_spec.js (hunk 1 above)
Comment (a-arias) [reply e2e/cypress/integration/messaging/search_group_message_spec.js]: ok makes sense, I will create the channel using the API. thanks

**Comment by a-arias** [comment]
@jasonblais  I just read the review and it totally makes sense, I will create a channel using the API and then assert the info. sorry for the delay.

**Comment by jasonblais** [comment]
Thanks @a-arias 👍

**Comment by hanzei** [comment]
Hey @a-arias,

Do you have any questions?

**Comment by a-arias** [comment]
@hanzei I asked in the chat about examples using the API, but I think we don't have it as far as I remember,.

I need some examples about how should I call the endpoints inside the tests.

any idea?

**Comment by saturninoabril** [comment]
@a-arias I'll look into this later.

**Comment by saturninoabril** [comment]
Did a few minor commits but should be good to merge once the build passed.

**Comment by saturninoabril** [comment]
@a-arias Thank you for your contribution!

**Pull request merged**